modules/mbedtls/configs/config-psa.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Generic configuration for PSA, manageable by Kconfig.
+ */
+
+#ifndef MBEDTLS_CONFIG_H
+#define MBEDTLS_CONFIG_H
+
+#define MBEDTLS_USE_PSA_CRYPTO
+#define MBEDTLS_PSA_CRYPTO_C
+
+/* enable built-in MAC */
+#define MBEDTLS_CIPHER_C
+
+#define MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG
+#define MBEDTLS_CMAC_C
+#define MBEDTLS_AES_C
+
+#define MBEDTLS_AES_ROM_TABLES
+
+#include "mbedtls/config_psa.h"
+#include "mbedtls/check_config.h"
```

Comment:
There is specific logic in these two files in the configuration header that should be used.
The file keeps the configuration definitions that the customer provides. There are a lot of them and most of them have explicit and implicit dependencies on each other.
There are, as I call them, "root" definitions like for example `MBEDTLS_CMAC_C`.
Header `config_psa.h` based on "root" definitions declares a lot of "subroot" definitions required to enable wished functionality.
Header `check_config.h` performs checking dependencies and generates compile time error if there is a lack of them or there are controversial settings.

I spent a quite long time figuring out which options should be used because including "subroot" options caused a lack of them and a request to add functionality that we really do not need.

I guess the current including as how it was intended to use by design.